src/elements/Button/Button.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import {
+	TouchableHighlight, FlexStyle, StyleProp, ViewStyle, TextStyle,
+} from 'react-native';
+import { ColorPalette, COLOR_SCHEMES, ColorScheme } from '@util/colorSchemes';
+import { useColorScheme } from 'react-native-appearance';
+import styles from './Button.styles';
+
+export type ButtonProps = {
+	onPress: Function; // Callback function called with the button is pressed.
+	children?: JSX.Element; // Elements to be wrapped by the button.
+	style?: StyleProp<FlexStyle>; // Dimension styling for the button root.
+	activeStyle?: StyleProp<ViewStyle>; // Styling for the button during an 'active' pseudo-state.
+	disabled?: boolean; // Whether the button can be interacted with.
+	palette?: Omit<ColorPalette, 'disabled'>; // Determines the color styling of the button.
+	compact?: boolean; // Whether the button should be without padding.
+	outlined?: boolean; // Whether the button is styled with an outline and transparent body.
+};
+
+export default ({
+	onPress,
+	children,
+	activeStyle, // TODO: test
+	style = {},
+	disabled = false,
+	palette = 'default',
+	compact = false, // TODO: test
```

Comment:
Should compact be a property? Or should compactness fall under the developer's responsibility to alter the padding and styling of the button.

Example of two compact buttons.
![Example of Compact Button](https://trello-attachments.s3.amazonaws.com/5db12c4da582b82f206fdb3e/5e64680d32bcd2540913aeef/019c9ddb9f28c771cdab8a860de66e92/Selection1.png)